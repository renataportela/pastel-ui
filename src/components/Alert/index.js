export { default, ALERT_COLORS } from './Alert'