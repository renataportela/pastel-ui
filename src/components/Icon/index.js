export { default, iconNames } from './Icon';
export { default as Svg } from './Svg';