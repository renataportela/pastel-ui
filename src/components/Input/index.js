export { default, INPUT_PROPS } from './Input'
export { inputStyles } from './ElementStyle'
