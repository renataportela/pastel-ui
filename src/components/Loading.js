import React from 'react'
import { colors, Icon, Overlay } from '~/pastel-ui'

function Loading(props){
  return (
    <Overlay>
      <Icon name="loading" width={30} fill={props.color || colors.secondary} />
    </Overlay>
  )
}

export default Loading