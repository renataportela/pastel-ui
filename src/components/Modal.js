import React, { useEffect, useRef, useState } from 'react';
import ReactDOM from 'react-dom';
import styled from '@emotion/styled';

import { scale } from '~/pastel-ui/theming';
import { Overlay, Paper } from '~/pastel-ui';
import Button from '~/pastel-ui/components/Button';
import useOutsideCloser from '~/pastel-ui/hooks/useOutsideCloser';

const modalRoot = document.getElementById('modal-root');

const Centered = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  z-index: 300;

  & > div {
    min-width: 300px;
    max-width: 90%;
    animation: ${scale} .1s;

    @media (min-width: 778px) {
      min-width: 600px;
    }
  }
`;

const CloseContainer = styled.div`
  float: right;
  margin-left: 20px;
  margin-bottom: 20px;
`;

const CloseButton = styled(Button)`
  position: absolute;
  top: 12px;
  right: 12px;
`;

function Modal(props){
  const [el] = useState(document.createElement('div'));
  const modalBoxRef = useRef();
  const close = () => props.setOpen(false);

  useOutsideCloser(modalBoxRef, close);

  useEffect(() => {
    modalRoot.appendChild(el);
    return () => modalRoot.removeChild(el);
  }, []);  

  if (props.open){
    return ReactDOM.createPortal(
        <Overlay>
          <Centered>
            <Paper ref={modalBoxRef}>
              <CloseContainer>
                <CloseButton 
                  pallete="primary" 
                  variant="outline" 
                  title={props.titleClose || 'Close'}
                  icon="close"
                  round
                  onClick={close}
                />
              </CloseContainer>
              {props.children}
            </Paper>
          </Centered>
        </Overlay>,
      el
    );
  }

  return null;
}

export default Modal;