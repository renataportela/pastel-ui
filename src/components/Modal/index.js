export { default } from './Modal'
export { default as ModalBase } from './ModalBase'