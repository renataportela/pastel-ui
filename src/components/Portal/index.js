export { default } from './Portal'
export { prepareClientPortals } from '@jesstelford/react-portal-universal'