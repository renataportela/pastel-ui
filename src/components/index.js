export { default as Alert } from './Alert'
export { default as AppBar } from './AppBar'
export { default as Avatar } from './Avatar'
export { default as BackTop } from './BackTop'
export { default as Badge } from './Badge'
export { default as Button, Close } from './Button'
export { default as Checkbox } from './Checkbox'
export { default as Chip } from './Chip'
export { default as Dialog } from './Dialog'
export { default as Drawer } from './Drawer'
export { default as Form } from './Form'
export { default as FormField } from './FormField'
export { Heading, Paragraph, Text } from './Typography'
export { default as Icon, Svg } from './Icon'
export { default as Menu } from './Menu'
export { default as Modal } from './Modal'
export { default as Overlay } from './Overlay'
export { default as PageLoading } from './PageLoading'
export { default as Pagination } from './Pagination'
export { default as Paper } from './Paper'
export { default as Portal } from './Portal'
export { default as Progress, CircularProgress } from './Progress'
export { default as Radio } from './Radio'
export { default as Tag } from './Tag'
export { default as TextField } from './TextField'
export { default as Theme, BaseColors, useTheme } from './Theme'
export { default as Toast, useToast, withToastProvider } from './Toast'
export { default as Ripple } from './Ripple'

export * from './Grid'