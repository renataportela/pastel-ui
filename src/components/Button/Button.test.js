import React from 'react'
import renderer from 'react-test-renderer'

import { Theme, Button } from '~/components'

describe('Button component', () => {
  it('renders a default button', () => {
    const tree = renderer
      .create(<Theme><Button label="Hello" /></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a danger button', () => {
    const tree = renderer
      .create(<Theme><Button label="Hello" color="danger" /></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a round button with menu icon', () => {
    const tree = renderer
      .create(<Theme><Button round icon="menu" /></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a loading button', () => {
    const tree = renderer
      .create(<Theme><Button label="Hello" loading /></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a small button', () => {
    const tree = renderer
      .create(<Theme><Button label="Hello" size="sm" /></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a large button', () => {
    const tree = renderer
      .create(<Theme><Button label="Hello" size="lg" /></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a extra large button', () => {
    const tree = renderer
      .create(<Theme><Button label="Hello" size="xl" /></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a outline button', () => {
    const tree = renderer
      .create(<Theme><Button label="Hello" color="secondary" kind="outline" /></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a ghost button', () => {
    const tree = renderer
      .create(<Theme><Button label="Hello" color="info" kind="ghost" /></Theme>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})