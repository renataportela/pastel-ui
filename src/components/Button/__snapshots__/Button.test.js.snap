// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders a danger button 1`] = `
<button
  className="ButtonStyle-sc-120zjl4-0 kIbNBe"
  color="danger"
  kind="default"
  size="md"
>
  Hello
</button>
`;

exports[`Button component renders a default button 1`] = `
<button
  className="ButtonStyle-sc-120zjl4-0 iJAzjI"
  color="primary"
  kind="default"
  size="md"
>
  Hello
</button>
`;

exports[`Button component renders a extra large button 1`] = `
<button
  className="ButtonStyle-sc-120zjl4-0 jJTQxI"
  color="primary"
  kind="default"
  size="xl"
>
  Hello
</button>
`;

exports[`Button component renders a ghost button 1`] = `
<button
  className="ButtonStyle-sc-120zjl4-0 kVStQG"
  color="info"
  kind="ghost"
  size="md"
>
  Hello
</button>
`;

exports[`Button component renders a large button 1`] = `
<button
  className="ButtonStyle-sc-120zjl4-0 kjIRiB"
  color="primary"
  kind="default"
  size="lg"
>
  Hello
</button>
`;

exports[`Button component renders a loading button 1`] = `
<button
  className="ButtonStyle-sc-120zjl4-0 iJAzjI"
  color="primary"
  disabled={true}
  kind="default"
  size="md"
>
  <svg
    className="Svg__SvgStyle-pyey0i-0 gYARyh"
    enableBackground="new 0 0 0 0"
    fill="none"
    height="24px"
    size={null}
    stroke="#8849bc"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    viewBox="0 0 512 512"
    width="24px"
    xmlSpace="preserve"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M288 39.056v16.659c0 10.804 7.281 20.159 17.686 23.066C383.204 100.434 440 171.518 440 256c0 101.689-82.295 184-184 184-101.689 0-184-82.295-184-184 0-84.47 56.786-155.564 134.312-177.219C216.719 75.874 224 66.517 224 55.712V39.064c0-15.709-14.834-27.153-30.046-23.234C86.603 43.482 7.394 141.206 8.003 257.332c.72 137.052 111.477 246.956 248.531 246.667C393.255 503.711 504 392.788 504 256c0-115.633-79.14-212.779-186.211-240.236C302.678 11.889 288 23.456 288 39.056z"
      fill="#8849bc"
    >
      <animateTransform
        attributeName="transform"
        attributeType="XML"
        dur="1s"
        from="0 256 256"
        repeatCount="indefinite"
        to="360 256 256"
        type="rotate"
      />
    </path>
  </svg>
  Hello
</button>
`;

exports[`Button component renders a outline button 1`] = `
<button
  className="ButtonStyle-sc-120zjl4-0 ipGwoh"
  color="secondary"
  kind="outline"
  size="md"
>
  Hello
</button>
`;

exports[`Button component renders a round button with menu icon 1`] = `
<button
  className="ButtonStyle-sc-120zjl4-0 dTtWyq"
  color="primary"
  kind="default"
  size="md"
>
  <svg
    className="Svg__SvgStyle-pyey0i-0 gYARyh"
    fill="none"
    height="24px"
    size={null}
    stroke="currentColor"
    strokeLinecap="round"
    strokeLinejoin="round"
    strokeWidth={2}
    viewBox="0 0 24 24"
    width="24px"
    xmlns="http://www.w3.org/2000/svg"
  >
    <line
      x1="3"
      x2="21"
      y1="12"
      y2="12"
    />
    <line
      x1="3"
      x2="21"
      y1="6"
      y2="6"
    />
    <line
      x1="3"
      x2="21"
      y1="18"
      y2="18"
    />
  </svg>
</button>
`;

exports[`Button component renders a small button 1`] = `
<button
  className="ButtonStyle-sc-120zjl4-0 bkToAt"
  color="primary"
  kind="default"
  size="sm"
>
  Hello
</button>
`;
